console-subscriber/src/lib.rs

```diff
@@ -304,3 +287,37 @@ impl proto::tasks::tasks_server::Tasks for Server {
         Ok(tonic::Response::new(stream))
     }
 }
+
+impl Visit for FieldsCollector {
+    fn record_debug(&mut self, field: &field::Field, value: &dyn std::fmt::Debug) {
+        self.fields.push(proto::Field {
+            name: Some(field.name().into()),
+            value: Some(value.into()),
+        });
+    }
+
+    fn record_i64(&mut self, field: &tracing_core::Field, value: i64) {
+        self.fields.push(proto::Field {
+            name: Some(field.name().into()),
+            value: Some(value.into()),
+        });
+    }
+    fn record_u64(&mut self, field: &tracing_core::Field, value: u64) {
+        self.fields.push(proto::Field {
+            name: Some(field.name().into()),
+            value: Some(value.into()),
+        });
+    }
+    fn record_bool(&mut self, field: &tracing_core::Field, value: bool) {
+        self.fields.push(proto::Field {
+            name: Some(field.name().into()),
+            value: Some(value.into()),
+        });
+    }
+    fn record_str(&mut self, field: &tracing_core::Field, value: &str) {
+        self.fields.push(proto::Field {
+            name: Some(field.name().into()),
+            value: Some(value.into()),
+        });
+    }
```

Comment:
note that [`&dyn std::error::Error` is also a potential primitive `tracing` field value](https://docs.rs/tracing/0.1.26/tracing/field/trait.Visit.html#method.record_error). 

this implementation will just record `Error`s with `fmt::Debug`, which is fine for now, but it might be nice to also allow recording `Error`s in a more structured way as a follow-up? we could have an `Error` message which includes the `Error` type's `fmt::Display` output and an optional [`source` field](https://doc.rust-lang.org/stable/std/error/trait.Error.html#method.source), or maybe the entire chain of `source`s as a protobuf list?

right now this wouldn't be particularly useful, since we only look at the `tokio::task` spans which never record `Error` fields. but in the future, we could potentially support nice error reporting in the console, so it's something that could be worth thinking about as a follow-up change...